drivers/pinctrl/pinctrl-amd.c

```diff
@@ -842,6 +842,32 @@ static const struct pinconf_ops amd_pinconf_ops = {
 	.pin_config_group_set = amd_pinconf_group_set,
 };
 
+static void amd_gpio_irq_init(struct amd_gpio *gpio_dev) {
+	struct pinctrl_desc *desc = gpio_dev->pctrl->desc;
+	unsigned long flags;
+	u32 pin_reg, mask;
+	int i;
+
+	mask = BIT(WAKE_CNTRL_OFF_S0I3) | BIT(WAKE_CNTRL_OFF_S3)
+		| BIT(INTERRUPT_MASK_OFF) | BIT(INTERRUPT_ENABLE_OFF)
+	        | BIT(INTERRUPT_MASK_OFF) | BIT(WAKE_CNTRL_OFF_S4);
+
+	for (i = 0; i < desc->npins; i++) {
+		int pin = desc->pins[i].number;
+		const struct pin_desc *pd = pin_desc_get(gpio_dev->pctrl, pin);
+		if (!pd)
+			continue;
+
+		raw_spin_lock_irqsave(&gpio_dev->lock, flags);
```

Comment:
Not sure if this locking is required or can maybe be avoided somehow since it's during probe. I think the pinctrl guys will tell you if that can be improved. For now it's better to be safe.

Reply:
They could probably be removed, if the registers were being modified at the same time as this operation that would probably mean something is configuring a pin, which would mean a possible race condition where a pin would be configured and then disabled here.

Reply:
Right, good point with the race condition.